redcap_bridge/utils.py

```diff
@@ -46,4 +49,34 @@ def get_repo_state(path):
         commit_hash = ''
         clean = False
 
-    return commit_hash, clean
\ No newline at end of file
+    return commit_hash, clean
+
+
+def compressed_record(csv_file, compressed_file=None):
+    custom_csv = pd.read_csv(csv_file)
+    df = pd.DataFrame(custom_csv)
+    custom_df = df.filter(regex='.___.')
+    for column in custom_df:
+        if 1 in custom_df[column].values:
+            style_moda = re.search('___(.+?)', column).group(1)
+            custom_df[column] = style_moda
+        else:
+            del custom_df[column]
+
+    list_column = custom_df.columns.tolist()
```

Comment:
I think you can simplify your life if you already here determine the final column names that will be present after the compression, e.g. via `modalities = set([x.split('___')[0] for x in list_column])` or a similar regex operation...

Then later on you only need to iterate over the modalities and can find all columns to be merged via `custom_df.filter(f'{modality___.')`